include/deal.II/non_matching/mapping_info.h

```diff
@@ -157,30 +326,544 @@ namespace NonMatching
+                              " active cell index"));
+            const auto it_begin =
+              unit_points.begin() +
+              unit_points_index
+                [cell_index_to_compressed_cell_index[cell_index]];
+            const auto it_end =
+              unit_points.begin() +
+              unit_points_index
+                [cell_index_to_compressed_cell_index[cell_index] + 1];
+            return make_array_view(it_begin, it_end);
+          }
+        else
+          {
+            const auto it_begin =
+              unit_points.begin() + unit_points_index[cell_index];
+            const auto it_end =
+              unit_points.begin() + unit_points_index[cell_index + 1];
+            return make_array_view(it_begin, it_end);
+          }
+      }
+    else if (cell_index != numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::faces_on_cells_in_vector,
+               ExcMessage("This mapping info is not reinitialized for faces"
+                          " on cells in a vector"));
+        if (do_cell_index_compression)
+          {
+            Assert(
+              cell_index_to_compressed_cell_index[cell_index] !=
+                numbers::invalid_unsigned_int,
+              ExcMessage(
+                "Mapping info object was not initialized for this active cell index"
+                " and corresponding face numbers"));
+            const unsigned int current_face_index =
+              cell_index_offset
+                [cell_index_to_compressed_cell_index[cell_index]] +
+              face_number;
+            const auto it_begin =
+              unit_points.begin() + unit_points_index[current_face_index];
+            const auto it_end =
+              unit_points.begin() + unit_points_index[current_face_index + 1];
+            return make_array_view(it_begin, it_end);
+          }
+        else
+          {
+            const unsigned int current_face_index =
+              cell_index_offset[cell_index] + face_number;
+            const auto it_begin =
+              unit_points.begin() + unit_points_index[current_face_index];
+            const auto it_end =
+              unit_points.begin() + unit_points_index[current_face_index + 1];
+            return make_array_view(it_begin, it_end);
+          }
+      }
+    else
+      AssertThrow(
+        false,
+        ExcMessage(
+          "cell_index has to be specified if face number is specified"));
+  }
+
+
+
+  template <int dim, int spacedim>
+  const typename MappingInfo<dim, spacedim>::MappingData &
+  MappingInfo<dim, spacedim>::get_mapping_data(
+    const unsigned int cell_index,
+    const unsigned int face_number) const
+  {
+    if (cell_index == numbers::invalid_unsigned_int &&
+        face_number == numbers::invalid_unsigned_int)
+      {
+        Assert(state == State::single_cell,
+               ExcMessage(
+                 "This mapping info is not reinitialized for a single cell"));
```

Comment:
```suggestion
                 "This mapping info is not reinitialized for a single cell!"));
```

Reply:
Done. I have added it to every Assert in the two getter functions.